grafanalib/core.py

```diff
@@ -1071,3 +1092,168 @@ def to_json_data(self):
             'valueMaps': self.valueMaps,
             'valueName': self.valueName
         }
+
+
+@attr.s
+class ColumnStyle(object):
+
+    alias = attr.ib(default="")
+    colorMode = attr.ib(default=None)
+    colors = attr.ib(default=[GREEN, ORANGE, RED])
+    dateFormat = attr.ib(default="YYYY-MM-DD HH:mm:ss")
+    decimals = attr.ib(default=2, validator=instance_of(int))
+    pattern = attr.ib(default="")
+    thresholds = attr.ib(default=attr.Factory(list))
+    type = attr.ib(default=NUMBER_COLUMN_STYLE)
+    unit = attr.ib(default=SHORT_FORMAT)
+
+    def to_json_data(self):
+        data = {
+            'alias': self.alias,
+            'pattern': self.pattern,
+            'type': self.type,
+        }
+        if self.type == DATE_COLUMN_STYLE:
+            data.update({
+                'dateFormat': self.dateFormat,
+            })
+        else:
+            data.update({
+                'colorMode': self.colorMode,
+                'colors': self.colors,
+                'decimals': self.decimals,
+                'thresholds': self.thresholds,
+                'unit': self.unit,
+            })
```

Comment:
Would you prefer I just do 4 different classes for the 4 column styles? It feels like that would be a lot of duplication (Since from the discussion it looks like inheritance would also be a bad idea)

Reply:
Not sure. Will think about it & get back to you.